src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,101 @@
+package duke;
+import java.io.IOException;
+
+public class Parser {
+    private TaskList tasks;
+    private Storage storage;
+    String dummyString;
+    Task dummyTask;
+    int counter;
+    int start;
+    int end;
+
+    public Parser(TaskList tasks, Storage storage) {
+        this.tasks = tasks;
+        this.storage = storage;
+    }
+
+    public void parse(String input) throws IOException {
+        try {
+            if (input.startsWith("mark")) {
+                dummyString = input.substring(5); //get number of task
+                counter = Integer.parseInt(dummyString) - 1;//convert to index of task (int)
+                tasks.get(counter).mark();
+                System.out.println("Nice! I've marked this task as done:\n" +
+                        "[" + tasks.get(counter).getStatusIcon() + "] " + tasks.get(counter).getDescription());
+                storage.saveFile(tasks);
+
```

Comment:
Perhaps u can remove the newline between `if` and `else if` statements etc? Same go with other lines of code.
```suggestion

```